src/components/Inventory/withdraw/flow/SelectPurposeForm.tsx

```diff
@@ -351,22 +352,24 @@ export default function SelectPurposeForm(props: SelectPurposeFormProps): JSX.El
       plantingSubzoneId: isOutplant ? localRecord.plantingSubzoneId : undefined,
       facilityId: Number(selectedNursery?.id || -1),
       batchWithdrawals: batches
-        .filter((batch) => {
+        .filter((batch: SearchResponseElement) => {
           return (
-            batch.facility_id.toString() === selectedNursery?.id.toString() &&
-            (!isOutplant || +batch['readyQuantity(raw)'] > 0)
+            `${batch.facility_id}` === selectedNursery?.id.toString() &&
+            (!isOutplant || Number(batch['readyQuantity(raw)']) > 0)
           );
         })
-        .map((batch) => ({
-          batchId: batch.id,
-          germinatingQuantityWithdrawn: isSingleOutplant ? 0 : isSingleBatch ? germinatingQuantityWithdrawn || 0 : 0,
-          notReadyQuantityWithdrawn: isSingleOutplant ? 0 : isSingleBatch ? notReadyQuantityWithdrawn || 0 : 0,
-          readyQuantityWithdrawn: isSingleOutplant
-            ? withdrawnQuantity || 0
-            : isSingleBatch
-            ? readyQuantityWithdrawn || 0
-            : 0,
-        })),
+        .map(
+          (batch: SearchResponseElement): BatchWithdrawalPayload => ({
+            batchId: Number(batch.id),
```

Comment:
Nice change, need to also update BatchWithdrawalFlow.tsxL129 (the comparison is failing, might need to convert them to Numbers before checking). The feature in question is this modal (which won't show in this PR but shows in staging). To trigger this modal, withdrawal all quantities as part of a withdrawal. 

<img width="277" alt="Terraware App 2023-12-20 11-01-14" src="https://github.com/terraware/terraware-web/assets/1865174/74e11207-115d-4312-b0e4-45435cdf2ecb">

Reply:
Thanks, I'll double check this functionality once it hits staging

Reply:
Works as expected in staging

Reply:
Awesome, thanks @nickgraz  🎉